plugins/basic/Python/tk_nuke_basic/plugin_bootstrap.py

```diff
@@ -298,14 +302,37 @@ def _report(self, progress_value, message):
         self._progress_task.report_progress(percentage, message)
         print message
 
-    def _on_finish(self):
+    def _on_finish(self, failed=False):
         """
         Called after bootstrap (success or failure) to cleanup resources.
         """
         # At this point we are guaranteed that bootstrapped is finished, so we can dismiss progress
         # reporting.
         self._progress_task.done()
 
+        # We should attempt to switch to the context that was supplied from the Nuke session that launched/spawned this
+        # session if we can. If this Nuke session has just been launched via the launch app this env var won't be set.
+        if not failed and self._previous_ctx_str:
+            # We have a context set previously, this must mean that this Nuke instance has been spawned from
+            # another which had TK context. We should switch to this context.
+            self._logger.debug("Context found from env var: \"TANK_CONTEXT\", context: %s" %  self._previous_ctx_str)
```

Comment:
[minor] Instead of concatenating the string yourself, you can let the log handler do it like below. That will keep Python from having to concat the string unless debug logging is turned on. In this case that's not going to be a performance issue, but in situation where there's a debug log message that's very high frequency, having to concat the string when it's not actually going to be logged (ie: debug logging is off) would be wasteful.

`self._logger.debug("Some message: %s that came from %s", (my_message, my_location))`